commit 43adf69062254fb4f8d4d11fb5fe36a60ae25d5a
Author: Ben Hutchings <ben@decadent.org.uk>
Date:   Sun Mar 14 18:41:56 2010 +0000

    Fix klibc Debian specific build trouble
    
    Simply remove the headers_install invocation and rely on linux-libc-dev.
    
    With these changes:
    
    (the extra chmod was needed to allow for umask > 022) I could compile,
    link and run this program:
    
    #include <unistd.h>
    
    int main(void)
    {
    	write(1, "Hello world\n", 12);
    	return 0;
    }
    
    Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
    Signed-off-by: maximilian attems <max@stro.at>

Index: b/scripts/Kbuild.install
===================================================================
--- a/scripts/Kbuild.install
+++ b/scripts/Kbuild.install
@@ -95,8 +95,14 @@
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
-	$(Q)$(MAKE) -C $(KLIBCKERNELSRC) ARCH=$(KLIBCARCH) INSTALL_HDR_PATH=$(INSTALLROOT)$(INSTALLDIR)/$(KCROSS) headers_install
+	$(Q)if [ -n "$(DEB_HOST_MULTIARCH)" ]; then \
+		ln -s /usr/include/$(DEB_HOST_MULTIARCH)/asm $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/ || exit; \
+	fi
+	$(Q)for x in /usr/include/linux /usr/include/asm*; do \
+		ln -s $${x} $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/ || exit; \
+	done
 	$(Q)cp -rf usr/include/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/.
+	$(Q)chmod -R a+rX $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)$(install-data) $(srctree)/klcc/klcc.1 $(INSTALLROOT)$(mandir)/man1/$(KCROSS)klcc.1
 	$(Q)$(install-bin) $(objtree)/klcc/$(KCROSS)klcc $(INSTALLROOT)$(bindir)
 
